wgpu-core/src/pipeline.rs

```diff
@@ -218,6 +229,32 @@ impl CreateShaderModuleError {
             _ => None,
         }
     }
+
+    pub fn main_range(&self) -> Option<std::ops::Range<usize>> {
```

Comment:
not entirely clear from the name that this is a source code range this error is about (how about `source_range`?) but more importantly, I'm confused about how this relates to `location` just above. Isn't that what we're actually looking for here?

Reply:
I added a separate method, since the existing `location` method requires a source string (and piping that one through is tough), and it computes a few pieces of information that I just end up throwing away. (line number and offset in the line)

If there's a better option that ultimately leads to simpler code, I'd love to hear about it.

Reply:
I see, makes sense!